src/yoroi-wallets/cardano/ShelleyWallet.ts

```diff
@@ -973,6 +987,32 @@ export class ShelleyWallet implements WalletInterface {
     return api.fetchCurrentPrice(symbol, this.getBackendConfig())
   }
 
+  async fetchNfts(): Promise<YoroiNFT[]> {
+    const utxos = this.utxos
+    const assets = utxos.flatMap((utxo) => utxo.assets ?? [])
+
+    const nftAssets = assets
+      .filter((asset) => asset.policyId && asset.name)
+      .map((asset) => {
+        const {policyId, name} = asset
+        return {nameHex: name, policy: policyId}
+      })
+
+    if (nftAssets.length === 0) {
+      return []
+    }
+
+    return api.getNFTs({assets: nftAssets}, this.getBackendConfig())
+  }
+
+  async fetchNftModerationStatus(fingerprint: string): Promise<YoroiNFTModerationStatus> {
+    const status = await api.getNFTModerationStatus(fingerprint, this.getBackendConfig())
+    if (status === YoroiNFTModerationStatus.PENDING) {
+      return new Promise((resolve) => setTimeout(() => resolve(this.fetchNftModerationStatus(fingerprint)), 5000))
```

Comment:
If the NFT is being checked the first time, it gets processed and responds with `YoroiNFTModerationStatus.PENDING` status. According to the documentation, we should be checking the response till the status changes. To prevent any huge network spike, I suggest we check this only twice at max and then refetch when the user drags down the list. What are your thoughts?

Reply:
It's breaking SRP, it should only get the status here, on UI if the user is waiting for this result, there we should refetch while it is present on the screen.